### PR TITLE
SC-4997 force user to change password if the password was changed by …

### PR DESCRIPTION
**Pull request by robertstrz**
…the admin

# Description

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-4997

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- controllers/forcePasswordChange.js  3
- static/scripts/dataprivacy/forcePasswordChange.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- controllers/forcePasswordChange.js  3
- static/scripts/dataprivacy/forcePasswordChange.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Complexity increasing per file
==============================
- controllers/forcePasswordChange.js  3
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
- static/scripts/dataprivacy/forcePasswordChange.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/dataprivacy/forcePasswordChange.js  2
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
- controllers/forcePasswordChange.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
- controllers/forcePasswordChange.js  3
- static/scripts/dataprivacy/forcePasswordChange.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

File: controllers/administration.js [hunk 1]
```diff
@@ -748,13 +748,23 @@ const returnAdminPrefix = (roles) => {
 };
 
 // with userId to accountId
-const userIdtoAccountIdUpdate = (service) => function useIdtoAccountId(req, res, next) {
+const userIdToAccountIdUpdate = (service) => async function useIdtoAccountId(req, res, next) {
```
Comment (CeEv) [inline controllers/administration.js:751]: please remove (service) and write 'accounts' in line 763

File: controllers/administration.js [hunk 2]
```diff
@@ -748,13 +748,23 @@ const returnAdminPrefix = (roles) => {
...
+	try {
+		await api(req)
+			.patch(`/users/${req.params.id}`, {
+				json: { forcePasswordChange: true },
```
Comment (CeEv) [inline controllers/administration.js:755]: Do you write test how the permission is work in server? Only person with permission in this school, superhero and the user him self should set this key.

File: controllers/administration.js (hunk 1 above)
Comment (robertstrz) [reply controllers/administration.js]: :+1:

File: controllers/administration.js (hunk 2 above)
Comment (robertstrz) [reply controllers/administration.js:755]: According to our discussion. No need to do that.

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- controllers/forcePasswordChange.js  3
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
- static/scripts/dataprivacy/forcePasswordChange.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- controllers/forcePasswordChange.js  3
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
- static/scripts/dataprivacy/forcePasswordChange.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/dataprivacy/forcePasswordChange.js  2
- controllers/forcePasswordChange.js  3
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by dominikbruechner** [comment]
just upfront, please always add screenshots of UI changes, thanks :)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
- controllers/forcePasswordChange.js  3
- static/scripts/dataprivacy/forcePasswordChange.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by robertstrz** [comment]
Screenshot

image

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/mocha/forcePasswordChange/forcePasswordChange.test.js  1
- static/scripts/dataprivacy/forcePasswordChange.js  2
- controllers/forcePasswordChange.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5748948&bid=18817138)

**Comment by robertstrz** [comment]
Reopen

**Pull request merged**